cpp/src/io/csv/csv_writer.cu

```diff
@@ -323,44 +327,65 @@ gdf_error write_csv(csv_write_arg* args)
         //
         // This is essentially an exclusive-scan (prefix-sum) across columns.
         // Moving left-to-right, add up each column and carry each value to the next column.
-        // Looks like we could transpose the matrix, scan it, and then untranspose it.
-        // Should be able to parallelize the math for this -- will look at prefix-sum algorithms.
+        // We do this in 3 kernel calls below.
         //
-        std::vector<char> buffer(memsize+1);
-        std::vector<size_t> string_locations(rows*count); // all the memory pointers for each column
+        rmm::device_vector<size_t> string_locations(rows*count); // all the memory pointers for each column
```

Comment:
The type usage for sizes is inconsistent in this function. Above you have `gdf_size_type` for the number of rows, `unsigned int` for the number of columns, and here you have `size_t`.

Reply:
The `gdf_size_type` is imposed by the column structure. The `unsigned int` for columns makes sense since it is a practical limit on the number of columns (4 billion) and should never be negative. The `size_t` is necessary because the string_locations values become memory offsets into what could be more than 4GB of memory.